tools/k8s-secret-syncer/resource_syncer.py

```diff
@@ -0,0 +1,89 @@
+import os
+import sys
+import json
+import subprocess
+import re
+import time
+import traceback
+
+# configuration
+RESOURCE_KIND = os.environ.get('RESOURCE_KIND', 'secret')  # cluster role must be granted accordingly
+SYNC_INTERVAL_SECONDS = int(os.environ.get('SYNC_INTERVAL_SECONDS', 300))
+SOURCE_NS = os.environ.get('SOURCE_NS', 'secrets')
+SOURCE_ANNO = os.environ.get('SOURCE_ANNO', 'ns-propagate')
+NS_BLACKLIST = os.environ.get('NS_BLACKLIST', 'kube-system,kube-public,default').split(',')
+SKIP_DELETE = os.environ.get('SKIP_DELETE', 'no') == 'yes'
+NS_BLACKLIST.append(SOURCE_NS)  # don't copy back to source
+
+def kube(kind, name, namespace=None, options={}, action='create'):
+  if namespace:
+    options['namespace'] = namespace
+  option_str = ' '.join([('--' + k + ' ' + v) for k,v in options.items()])
+  return subprocess.call(["/bin/sh", "-c", "kubectl {} {} {}".format(action, kind, name) + ' ' + option_str])
+
+def kube_get(kind, namespace=None, name=None):
+  cmd = 'kubectl get -o json ' + kind + (' -n ' + namespace if namespace else ' ') + (name if name else ' ')
+  resp = subprocess.check_output(['/bin/sh', '-c', cmd])
+  return json.loads(resp)['items']
+
+def kube_apply(definition):
+  return subprocess.call(["/bin/sh", "-c", "echo '" + json.dumps(definition) + "' | kubectl apply -f -"])
```

Comment:
Instead of using the shell to echo the definition, you should use:

```python
ps = subprocess.Popen(["/usr/bin/kubectl", "apply", "-f", "-"], stdin=subprocess.PIPE)
ps.communicate(json.dumps(definition))
ps.wait()
```